### PR TITLE
Additional payment method for membership fees

### PR DESCRIPTION
**Pull request by KaiNissen**
This PR depends on wmde/FundraisingStore#99. Build will fail until that is merged.

Still missing:
- IPN handler for the actions that are triggered on creating a subscription and receiving a payment
- Daily data export
- Changing the confirmation page and the confirmation mail template (wmde/fundraising#1093)

### CODE REVIEW

**Comment by gbirke** [comment]
The code looks good, but I'd like to try it out in the browser. Could you update the templates in the CMS?

File: tests/Unit/PayPalUrlGeneratorTest.php [hunk 1]
```diff
@@ -86,11 +101,58 @@ private function newIncompletePayPalUrlConfig() {
...
+		$this->assertContains( 'no_shipping=1', $generatedUrl );
+		$this->assertContains( 'src=1', $generatedUrl );
+		$this->assertContains( 'sra=1', $generatedUrl );
+		$this->assertContains( 'srt=0', $generatedUrl );
+		$this->assertContains( 'a1=0', $generatedUrl );
+		$this->assertContains( 'p1=90', $generatedUrl );
+		$this->assertContains( 't1=D', $generatedUrl );
+		$this->assertContains( 'a3=12.34', $generatedUrl );
+		$this->assertContains( 'p3=3', $generatedUrl );
+		$this->assertContains( 't3=M', $generatedUrl );
```
Comment (JeroenDeDauw) [inline tests/Unit/PayPalUrlGeneratorTest.php]: A bunch of this stuff is repeated in other tests. Also, it's not apparent what the purpose of all these things is. Are all of these things critical to us, or are some of them things that just normally are in there? Are there logical groups that can be extracted into something like `assertContainsPaymentIntervalParameters`?

File: tests/Data/ValidMembershipApplication.php [hunk 2]
```diff
@@ -168,13 +195,33 @@ private static function createDoctrineApplicationWithoutApplicantName(): Doctrin
...
+
+		$application->setMembershipType( self::MEMBERSHIP_TYPE );
+		$application->setPaymentIntervalInMonths( self::PAYMENT_PERIOD_IN_MONTHS );
+		$application->setPaymentAmount( self::PAYMENT_AMOUNT_IN_EURO );
+
+		$application->setPaymentBankAccount( self::PAYMENT_BANK_ACCOUNT );
+		$application->setPaymentBankCode( self::PAYMENT_BANK_CODE );
+		$application->setPaymentBankName( self::PAYMENT_BANK_NAME );
+		$application->setPaymentBic( self::PAYMENT_BIC );
+		$application->setPaymentIban( self::PAYMENT_IBAN );
```
Comment (JeroenDeDauw) [inline tests/Data/ValidMembershipApplication.php]: Did you accidentally undo a commit I made here or is this on purpose?

**Comment by JeroenDeDauw** [comment]
Same here, generally looks good; apart from the presumably accidental revert

**Comment by KaiNissen** [comment]
CMS template updated for the test system.

File: tests/Data/ValidMembershipApplication.php (hunk 2 above)
Comment (KaiNissen) [inline tests/Data/ValidMembershipApplication.php]: Oh, err, yeah. I did.

**Comment by gbirke** [comment]
I've tested in the browser. It works. 

There are some cosmetic issues. I fixed one of them, the other I'm not sure if this HTML code can be removed: 

```html
<div style="margin-left: 180px" class="input-container">
	<span id="bank-name"></span><br><span id="bic"></span>
</div>

<hr>
<div class="last direct-debit label"></div>
```

Can you look into that?

**Comment by KaiNissen** [comment]
`<span id="bank-name"></span>` is used for displaying the bank name determined by the bank data validator.

`<span id="bic"></span>` seems to be a leftover from the times we didn't even ask for the BIC, because we also determined that automatically.

What the last div in your snippet is about I can't even guess. There is nothing in neither the browser tests nor the old application referencing that (apart from a style definition). I just removed it from the template. Will remove the style definition as well.

**Comment by JeroenDeDauw** [comment]
+1

**Comment by JeroenDeDauw** [comment]
@gbirke such merge?

**Pull request merged**